src/tree.jl

```diff
@@ -113,6 +113,6 @@ elements of type specified by the 3rd argument. (This needs to be specified
 explicitly, because the typesystem generally cannot guess the universal type
 correctly.)
 """
-tree_map(x::Tree, f, ::Type{T}) where {T} = Tree{T}(k => tree_map(v, f, T) for (k, v) in x)
+tree_map(f, x::Tree, ::Type{T}) where {T} = Tree{T}(k => tree_map(f, v, T) for (k, v) in x)
```

Comment:
good, more in line with map functions I think

Reply:
yeah also allows the `do` notation

```
map(mytree, Int) do c
  whatever(c.value .....)
 end
```

Reply:
ok this is demonstrated in docs now, all ok.